html/modules/custom/ghi_menu/src/Form/MenuLinkDeleteWithSublinksForm.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\ghi_menu\Form;
+
+use Drupal\Core\Entity\ContentEntityDeleteForm;
+use Drupal\menu_link_content\Entity\MenuLinkContent;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Deletes a menu link with its Sublinks.
+ */
+class MenuLinkDeleteWithSublinksForm extends ContentEntityDeleteForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $entity = $this->getEntity();
+    $config_name = MenuLinkDeleteWithSublinksSettingsForm::CONFIG_NAME;
+    $enabled_menus = \Drupal::config($config_name)->get('enabled_menus') ?? [];
+
+    if ($entity instanceof MenuLinkContent && in_array($entity->getMenuName(), $enabled_menus)) {
+      $this->deleteSublinks($entity);
+    }
+
+    parent::submitForm($form, $form_state);
```

Comment:
Also, if there is an easy enough way to _first_ delete the parent and to delete the sublinks only if that deletion was successful, I think that would be a bit safer. Not sure if it's technically feasible though.